### DISCUSSION TITLE
Getting all subgroups of a group

### DISCUSSION THREAD

**Discussion by Moult** [HDFGroup/hdf5#4034] [Q&A]
G'day all! I'm a C++ beginner so perhaps I'm missing something obvious, but this is some code I'm writing to loop through all the subgroups of a group.

```cpp
            H5::Group shapes_g = file.openGroup("shapes");
            hsize_t total_shapes = shapes_g.getNumObjs();

            for (hsize_t i = 0; i < total_shapes; ++i) {
                std::string shapeName = shapes_g.getObjnameByIdx(i);
                H5::Group shapeGroup = shapes_g.openGroup(shapeName);
                // ... etc
            }
```

When profiling, I've found that `shapes_g.getObjnameByIdx(i)` is very slow and actually makes my execution time significantly higher (e.g. 5x). Is there a more efficient way to loop through subgroups? In some of the sample data I'm working with there are 60,000 subgroups.

**Comment by derobins** [comment]
Can you post this on the forum? GitHub Discussions are for new features, etc.

https://forum.hdfgroup.org/
